src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,6 +392,161 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
 };
 
 
+/*
+  This class handles the neural net computation in parallel.
+
+  It can accept just input features, or input features plus iVectors.  */
+class BatchNnetComputer {
```

Comment:
It can be friend of class DecodableNnetSimple to have no need to copy some functions!

Reply:
If I want class BatchNnetComputer use some functions of class DecodableNnetSimple. I should declare "friend class BatchNnetComputer" in class DecodableNnetSimple. But I think DecodableNnetSimple is a footstone of many other classes, such as DecodableAmNnetSimple. So I don't want to modify it actually.
Otherwise, only the function "CheckAndFixConfigs()" is copied. So I didn't do as your comment before.